deluge/ui/gtkui/connectionmanager.py

```diff
@@ -233,6 +234,23 @@ def add_host(self, host, port, username='', password=''):
         # Update the status of the hosts
         self.__update_list()
 
+    def on_entry_host_paste_clipboard(self, widget):
+        paste = gtk.clipboard_get().wait_for_text()
+        log.debug('on_entry_proxy_host_paste-clipboard: got paste: %s', paste)
+        if '//' not in paste:
+            parsed = urlparse('//' + paste)
```

Comment:
If there is no scheme (like `http://`), the address needs an empty scheme (or "prefixing with slashes") so that eg. ?in `username:password@host` style addresses, `username` does not get confused with a scheme identifier.